lib/wallet/rpc.js

```diff
@@ -559,18 +578,37 @@ class RPC extends RPCBase {
     const height = this.wdb.state.height;
 
     const hash = parseHash(addr, this.network);
-    const txs = await wallet.getHistory();
+
+    // While this doesn't consume a lot of memory
+    // it can still consume a lot of CPU and be very
+    // slow. If this is a common information to query,
+    // it would be better to calculate the total per
+    // address while indexing and adding blocks.
+
+    let txs = await wallet.listHistory(null, {
+      limit: 5,
+      reverse: false
+    });
 
     let total = 0;
 
-    for (const wtx of txs) {
-      if (wtx.getDepth(height) < minconf)
-        continue;
+    while (txs.length) {
+      for (const wtx of txs) {
+        if (wtx.getDepth(height) < minconf)
+          continue;
 
-      for (const output of wtx.tx.outputs) {
-        if (output.getHash().equals(hash))
-          total += output.value;
+        for (const output of wtx.tx.outputs) {
+          const ohash = output.getHash();
+          if (ohash && ohash.equals(hash))
+            total += output.value;
+        }
       }
+
+      txs = await wallet.listHistoryAfter(null, {
+        hash: txs[txs.length - 1].hash,
```

Comment:
Seeing this I worry it is possible to reference `.hash` on `undefined` or `null` in some edge case

Reply:
It's within a `while (txs.length) {`, so txs has to have a length here. The result from `listHistoryAfter` is always an array.